--- a/sim/common/nrun.c
+++ b/sim/common/nrun.c
@@ -67 +67 @@ main (int argc, char **argv)
-  RETSIGTYPE (*prev_sigint) ();
+  RETSIGTYPE (*prev_sigint) (int);
--- a/sim/arm/armdefs.h
+++ b/sim/arm/armdefs.h
@@ -267 +267 @@
-				 unsigned (*func) ());
+				 unsigned (*func) (ARMul_State *));
--- a/sim/common/dv-sockser.c
+++ b/sim/common/dv-sockser.c
@@ -178 +178 @@ dv_sockser_init (SIM_DESC sd)
-    RETSIGTYPE (*orig) ();
+    RETSIGTYPE (*orig) (int);
